File: RESUME_APP/frontend/js/cover-letter.js
// Cover letter generation functionality
async function generateCoverLetter() {
    const jobDescription = document.getElementById('coverLetterJobDesc').value;
    const name = document.getElementById('applicantName').value || 'Your Name';
    const tone = document.getElementById('coverLetterTone').value;
    
    if (!jobDescription.trim()) {
        alert('Please enter a job description');
        return;
    }
    
    const resumeData = getResumeData();
    
    if (!resumeData.text) {
        alert('Please upload a resume first');
        showSection('upload');
        return;
    }
    
    showLoading();
    
    try {
        const result = await apiCall('/generate-cover-letter', 'POST', {
            resume_text: resumeData.text,
            job_description: jobDescription,
            tone: tone,
            name: name
        });
        
        displayCoverLetter(result.cover_letter);
        
    } catch (error) {
        alert('Error generating cover letter: ' + error.message);
        console.error(error);
    } finally {
        hideLoading();
    }
}

function displayCoverLetter(coverLetter) {
    const resultDiv = document.getElementById('coverLetterResult');
    resultDiv.textContent = coverLetter;
    resultDiv.style.display = 'block';
    
    // Add download button
    const downloadBtn = document.createElement('button');
    downloadBtn.className = 'btn btn-primary';
    downloadBtn.textContent = 'Download as Text';
    downloadBtn.style.marginTop = '1rem';
    downloadBtn.onclick = () => {
        const blob = new Blob([coverLetter], { type: 'text/plain' });
        const url = URL.createObjectURL(blob);
        const a = document.createElement('a');
        a.href = url;
        a.download = 'cover_letter.txt';
        a.click();
        URL.revokeObjectURL(url);
    };
    
    if (!resultDiv.querySelector('button')) {
        resultDiv.appendChild(downloadBtn);
    }
}

File: RESUME_APP/backend/app/routes/cover_letter.py
from flask import Blueprint, request, jsonify
from backend.app.services.cover_letter_generator import CoverLetterGenerator

bp = Blueprint('cover_letter', __name__, url_prefix='/api')

@bp.route('/generate-cover-letter', methods=['POST'])
def generate_cover_letter():
    """Generate a personalized cover letter."""
    data = request.get_json()
    resume_text = data.get('resume_text', '')
    job_description = data.get('job_description', '')
    tone = data.get('tone', 'formal')
    name = data.get('name', 'Your Name')
    
    if not resume_text or not job_description:
        return jsonify({'error': 'Resume text and job description are required'}), 400
    
    try:
        generator = CoverLetterGenerator()
        cover_letter = generator.generate_cover_letter(
            resume_text, 
            job_description, 
            tone=tone,
            name=name
        )
        
        return jsonify({
            'success': True,
            'cover_letter': cover_letter
        })
    
    except Exception as e:
        return jsonify({'error': str(e)}), 500

